src/store/movies-slice.ts

```diff
@@ -1,50 +1,84 @@
 import { PayloadAction } from '@reduxjs/toolkit';
-import { Movie } from '../movie';
+import { Movie, MoviesRes } from '../movie';
 import axios from 'axios';
 import { AppState } from './reducer';
 
+const config = {
+    baseURL: 'http://localhost:4000',
+}
+
 export interface MoviesState {
     movies: Movie[];
     totalAmount: number;
+    newMovie: Movie;
 }
 
 const initialState: MoviesState = {
     movies: [],
-    totalAmount: 0
+    totalAmount: 0,
+    newMovie: null
 };
 
+export const setMovies = (data: MoviesRes) => ({
+    type: 'movies/setMovies',
+    payload: data
+})
+
+export const movieAdded = (movie: Movie) => ({
+    type: 'movies/movieAdded',
+    payload: movie
+})
+
+export const fetchMovies = () => async (dispatch: any, getState: any) => {
+    const state: AppState = getState();
+    const { genre, ...rest } = state.filters;
+    const filter = genre;
+
+    const res: { data: MoviesRes } = await axios.get('/movies', {
+            ...config, params: { ...rest, filter }
+        }
+    );
+
+    dispatch(setMovies(res.data));
+}
+
+export const addMovie = (movie: Movie) => async (dispatch: any) => {
+    const res: { data: Movie } = await axios.post('/movies', movie, config);
+
+    dispatch(movieAdded(res.data));
+    dispatch(fetchMovies());
+}
+
+export const editMovie = (movie: Movie) => async (dispatch: any) => {
+    await axios.put('/movies', movie, config);
+
+    dispatch(fetchMovies());
+}
+
+export const removeMovie = (id: string) => async (dispatch: any) => {
+    await axios.delete(`/movies/${id}`, config);
+
+    dispatch(fetchMovies());
+}
+
 export function moviesReducer(state = initialState, action: PayloadAction<any>) {
     switch (action.type) {
         case 'movies/setMovies': {
             const { data, totalAmount } = action.payload;
             return { ...state, movies: data, totalAmount }
         }
 
-        // case 'movies/addMovie': {
-        //     return [ ...state, action.payload ]
-        // }
-        //
-        // case 'movies/editMovie': {
-        //     const { id, data } = action.payload;
-        //     return state.map(movie => (movie.id === id) ? data : movie);
-        // }
-        //
-        // case 'movies/deleteMovie': {
-        //     return state.filter(m => m.id !== action.payload);
-        // }
+        case 'movies/movieAdded': {
```

Comment:
Present a good practice to name actions type in UPPER_SNAKE_CASE

Reply:
NP, usually do like this, but in redux official documentation they wrote with camel case